### PR TITLE
PHPCS-68 Refactored some tag sniffs into separate files

### PR DESCRIPTION
**Pull request by b3nl**
PHPCS-68 removed the tag validation thru the factory and the abstract
PHPCS-68 Added an exception mechanism for the errors and warnings
PHPCS-68 Extended the doc with new infos for development and testing

### CODE REVIEW

File: src/Standards/BestIt/CodeSniffer/Helper/DocTagHelper.php [hunk 1]
```diff
@@ -42,20 +40,6 @@ class DocTagHelper
      */
     private $tokens;
 
-    /**
```
Comment (b3nl) [inline src/Standards/BestIt/CodeSniffer/Helper/DocTagHelper.php]: Now dead codes. Was "redundant" architecture anways.

File: src/Standards/BestIt/CodeSniffer/Helper/DocTagOccurenceHelper.php [hunk 2]
```diff
@@ -1,105 +0,0 @@
-<?php
```
Comment (b3nl) [inline src/Standards/BestIt/CodeSniffer/Helper/DocTagOccurenceHelper.php]: Now dead codes. Was "redundant" architecture anways.

File: src/Standards/BestIt/CodeSniffer/Helper/DocTagHelper.php (hunk 1 above)
Comment (b3nl) [inline src/Standards/BestIt/CodeSniffer/Helper/DocTagHelper.php]: No dead codes. Was "unneeded architecture" anyways.

File: src/Standards/BestIt/Sniffs/AbstractSniff.php [hunk 3]
```diff
@@ -23,14 +23,14 @@
      *
      * @var File|void
      */
-    private $file;
+    protected $file;
```
Comment (b3nl) [inline src/Standards/BestIt/Sniffs/AbstractSniff.php]: We chose to move away from getters and setters!

File: src/Standards/BestIt/Sniffs/Commenting/AbstractDocSniff.php [hunk 4]
```diff
@@ -181,48 +180,6 @@
      */
     public const MESSAGE_DESCRIPTION_TOO_LONG = 'The description exceeds the maximum length of 120 chars.';
 
-    /**
```
Comment (b3nl) [inline src/Standards/BestIt/Sniffs/Commenting/AbstractDocSniff.php]: No dead codes. Was "unneeded architecture" anyways.

File: src/Standards/BestIt/Sniffs/DocTags/AbstractTagSniff.php [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+use function implode;
+use function spl_autoload_call;
+use function substr;
+use const T_DOC_COMMENT_CLOSE_TAG;
+use const T_DOC_COMMENT_STRING;
+use const T_DOC_COMMENT_TAG;
+
+// Get the custom constants.
+if (!class_exists(Tokens::class, false)) {
+    spl_autoload_call(Tokens::class);
```
Comment (migo315) [inline src/Standards/BestIt/Sniffs/DocTags/AbstractTagSniff.php]: (thumbsup)

**Pull request merged**